### PR TITLE
[irods#4626] Added new icommand: istream (master)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: src/istream.cpp [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+auto usage() -> void
+{
+    std::cout << "Usage: istream read [-o INTEGER] [-c INTEGER] LOGICAL_PATH\n"
+                 "Usage: istream write [-o INTEGER] [-c INTEGER] [--no-trunc] [-a] LOGICAL_PATH\n"
+                 "\n"
+                 "Streams bytes to/from iRODS via stdin/stdout.\n"
+                 "Reads bytes from the target data object and prints them to stdout.\n"
+                 "Writes bytes from stdin to the target data object.\n"
+                 "\n"
+                 "If the target data object does not exist and --no-trunc is not passed,\n"
```
Comment (trel) [inline src/istream.cpp]: i believe this is correct - just a very confusing triple negative...  let's play with the wording a bit.

**Comment by trel** [comment]
and of course, some tests are required.

**Comment by korydraughn** [comment]
The formatting needs to be adjusted for the man page too. Not sure what the rules are around that.

**Comment by trel** [comment]
the man pages are auto-generated by help2man.   As long as the main `-h` option prints cleanly, we're good for now.

**Comment by trel** [comment]
i like it.   anything else to do here?

**Pull request merged**